meshroom/ui/qml/ImageGallery/ImageGallery.qml

```diff
@@ -201,6 +200,17 @@ Panel {
                 }
             }
 
+            // Update grid item when corresponding thumbnail is computed
+            Connections {
+                target: ThumbnailCache
+                function onThumbnailCreated(imgSource, callerID) {
+                    let item = grid.itemAtIndex(callerID);  // item is an ImageDelegate
+                    if (item && item.source == imgSource) {
+                        item.updateThumbnail();
+                    }
```

Comment:
Maybe add an else with a full search of the imgSource, as if the filter is enabled the filteredIndex changed and the item is never updated.
After a night, the busyindicator is still running.